Plugins/Animation/src/AnimationComponent.hpp

```diff
@@ -18,39 +18,87 @@ namespace AnimationPlugin {
 
 class SkeletonBoneRenderObject;
 
+/// The AnimationComponent is responsible for the management of the animations of an Entity.
+/// It stores the animation Skeleton and the animation data as well as the animation playing speed.
```

Comment:
So here, `Animation` means Skeletal Animation (or equivalent).
This cannot be used to animate an object (e.g. camera, position) ?

Reply:
Update: after reading the rest of the code.
The animation component seems to be in charge of many other things, like creating the skeleton bones for display.
Please clearly state the role of this object.

Reply:
I would say, without taking a deep look at it, it should be renamed AnimationSkeletonComponent, though I have to investigate about its use for other animation purposes (e.g. camera, position, ...)

Reply:
For now, the `AnimationPlugin` only deals with skeleton-based character animation.
Two choices: 
 - just specify this in the doc for now, and improve the animation management in another PR.
 - include it in this PR, but then postpone its review/acceptance to let other PRs pass.

Regarding the `AnimationComponent`, I don't really know how to improve it. A possible solution would be the following.
Let the `AnimationComponent` only store the `Core::Animation`s and current animation id and time. It will also expose the current animation and pose.
Then there could be an `AnimationSkeletonComponent` which would only store the `Core::Animation::Skeleton` and a map from bone id to transform storage index in the animation pose. It would also be responsible for updating the skeleton pose, through Component IO, and display it.

But I'm not sure about it, and if we want to handle other animation purposes, we need a deep reflection on how to deal with such cases, i.e. how to update the animated data.
Let's say we have a cube, with keyframed transforms. Then the `AnimationComponent` might store all the transforms, but how to / who should change the cube's transform when playing the animation? And what about animated cameras?
We need a mechanism to bind the animation transform to its use.